service/storage_proxy.cc

```diff
@@ -3679,9 +3679,12 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
         make_mutation_data_requests(cmd, data_resolver, _targets.begin(), _targets.end(), timeout);
 
         // Waited on indirectly.
-        (void)data_resolver->done().then(utils::result_into_future<result<>>).then_wrapped([this, exec, data_resolver, cmd = std::move(cmd), cl, timeout] (future<> f) {
-            try {
-                f.get();
+        (void)data_resolver->done().then_wrapped([this, exec, data_resolver, cmd = std::move(cmd), cl, timeout] (future<result<>> f) {
+            (void)utils::result_try([&] () -> result<> {
```

Comment:
Why the cast to void? I guess bo::result is [[nodiscard]]?

Every cast to void deserves a common explaining why it is justified.

Reply:
(the [[nodiscard]] isn't the justification, but that all errors are handled is).

Reply:
Yes, bo::result is nodiscard. I'll add comments with explanations.

I also had an idea to change result_try so that it does not return a result if it always handles all exceptions, i.e. it has catch_dots and none of the catch clauses return exceptions, but it's definitely out of scope of this PR (also, not sure if it wouldn't be too confusing).

Reply:
Here I am also not sure if `result_try` is a simplification. You check for `res` to be an error explicitly anyway.

Reply:
@gleb-cloudius Won't explicit checking cause duplication of the exception handler? Although the handler is very simple, it is still duplication and you will have to remember about both of them if you want to introduce changes.

To be clear, I'm interpreting your suggestion as something like this:

```c++
try {
    result<> res = f.get();
    if (!res) {
        // Handler no. 1
        _result_promise.set_value(std::move(res).error());
        return;
    }

    /* ... lots of code here ... */

} catch (...) {
    // Handler no. 2
    _result_promise.set_exception(std::current_exception());
}
```

Reply:
My point is that you already do explicit checking and then you re-check once again in the `try` wrapper. The idea of the `try` wrappers was to hide the diff between an error result and an exception, but here you forced to deal with the result anyway. I do not think resolving `_result_promise` twice is a big issue.

Reply:
Although I have to check the result explicitly, I'm not doing any real error handling there and just return it so that `result_try` handles it. The idea of the `result_try` helpers was not only to make commits easier to read, but also to handle both thrown exceptions and failed results in a unified way - you only need to write the `catch` handler once and it will work for both ways of reporting exceptions.

I won't insist on it too much - I can change error handling here in the way you wish - but I think there is a benefit to keeping the error logic defined in one place.

Reply:
Done, converted this code so that the `_result_promise` is set in two places.

Reply:
Added a comment for the void cast, too.